src/Airship/Internal/Route.hs

```diff
@@ -65,6 +102,42 @@ var t = Route [Var t]
 star :: Route
 star = Route [RestUnbound]
 
+
+-- Routing trie creation algorithm
+-- 1. Store full paths as keys up to first `var`
+-- 2. Take SHA256 digest of the URL portion preceding the
```

Comment:
Since you're only using the hash for a delimiter that the part of the path is a variable, you could instead use delimiter character. Rather than trying to find a printable character that's unused, I'd recommend using something like `'\US' :: Char`, which is the non-printable, ASCII Unit Separator character. `BC8.singleton '\US'`